vignettes/formats_precedence.Rmd

```diff
@@ -0,0 +1,300 @@
+```
+cell <- parent_table <- parent_table <- ... <- parent_table
+```
+In case when the format is specified by the user in more than one place, the one which is closer to the `cell` in the precedence-inheritance chain will be applied. In the following sections of this vignette, we illustrate the format precedence rules with a few examples.
+
+### Standard format
+
+Below is a simple layout that does not explicitly set any specific format for the output of the analysis function used in this example. In such a case, the default format will be used.
+
+```{r}
+lyt0 <- basic_table() %>%
+  split_cols_by("ARM") %>%
+  analyze(vars = "AGE", afun = mean)
+
+build_table(lyt0, ADSL)
+```
+
+### Cell format
+
+The format of the cell can be explicitly specified through `rcell()` or `in_rows()` functions. The former is essentially a collection of data objects while the latter is a collection of `rcell()` objects. In the context of inheritance, this is the most specific place where the format can be specified by the user.
+
+```{r}
+lyt1 <- basic_table() %>%
+  split_cols_by("ARM") %>%
+  analyze(vars = "AGE", afun = function(x) {
+    rcell(mean(x), format = "xx.xx", label = "Mean")
+  })
+
+build_table(lyt1, ADSL)
+
+lyt1a <- basic_table() %>%
+  split_cols_by("ARM") %>%
+  analyze(vars = "AGE", afun = function(x) {
+    in_rows(
+      "Mean" = rcell(mean(x)),
+      .formats = "xx.xx"
+    )
+  })
+
+build_table(lyt1a, ADSL)
+```
+
+If the format is specified in both of these places at the same time, the one specified through `in_rows()`, has higher precedence. Technically, in this case, the format defined in `rcell()` will simply be overwritten by the one defined in `in_rows()`.
+
+```{r}
+lyt2 <- basic_table() %>%
+  split_cols_by("ARM") %>%
+  analyze(vars = "AGE", afun = function(x) {
+    in_rows(
+      "Mean" = rcell(mean(x), format = "xx.xxx"),
+      .formats = "xx.xx"
+    )
+  })
+
+build_table(lyt2, ADSL)
+```
+
+### Parent table format and inheritance
+
+In addition to the cell level, the format can be specified at the parent table level(s). Then, if no format has been set by the user for a cell, the format from the nearest parent split table will be inherited (if any).
+
+```{r}
+lyt3 <- basic_table() %>%
+  split_cols_by("ARM") %>%
+  analyze(vars = "AGE", mean, format = "xx.x")
+
+build_table(lyt3, ADSL)
+```
+
+If the cell format was specified too, then the parent table format is ignored for this cell, since the cell format is treated as the most specific.
+
+```{r}
+lyt4 <- basic_table() %>%
```

Comment:
Probably don't need this example, just the one below it that shows a cell that does inherit and a cell that doesn't  coming from the same analyze block

Reply:
Thanks @gmbecker 
Hmm, to be honest I find layouts 4 and 4a valuable to this vignette, as they are somehow in between layouts 3 and 5. So I would prefer to keep it. However, if you sure that they should be removed - let me know and I will do it. Tx!

Reply:
4 doesn't convey any information that 4a doesn't also communicate equally well. We don't want documentation to be longer than it needs to be because there's already a lot of it and there will be more in the future. @wwojciech

Reply:
I will remove it then. Tx for additional argumentation.